File: sort/selection-sort.js
let unsorted = [5, 2, 9, 1, 11, 7, 8, -1, 4, 3, 10, 14, 0, 6];

const selection_sort = (unsorted) => {

    for (let outer = 0; outer < unsorted.length - 1 ; outer++) {
        
        var min = outer;

        for (let inner = outer + 1; inner < unsorted.length; inner++) {
            if (unsorted[min] > unsorted[inner]) {
                min = inner;
            }
        }

        let temp = unsorted[outer];
        unsorted[outer] = unsorted[min];
        unsorted[min] = temp;
    }

    console.log(unsorted);
}

selection_sort(unsorted);

File: sort/SelectionSort.java
package Sort;

public class SelectionSort {
   
   public static void main(String[] args) {

      int[] unsorted = { 3, 6, 2, 5, 9, 10, 7, 4, 8, 1 };

      int[] sorted = selectionSort(unsorted);
      
      for (int i : sorted) {
         System.out.println(i);
      }
   }
   
   public static int[] selectionSort(int[] unsorted) {

      for (int i = 0; i < unsorted.length; i++) {

         int min = i;

         for (int j = i + 1; j < unsorted.length; j++) {
            if (unsorted[j] < unsorted[min]) {
               min = j;
            }
         }

         int tmp = unsorted[i];
         unsorted[i] = unsorted[min];
         unsorted[min] = tmp;
      }
      
      return unsorted;
   }
}
